### ISSUE TITLE
Error Using NanoGUI with GLFW(c++)

### ISSUE THREAD

**Issue by nikhilanayak** [wjakob/nanogui#454]
I recently started trying to use NanoGUI with GLFW and I keep getting an error.
The specific lines where this error comes from are these:
```
nanogui::Screen *screen = new nanogui::Screen;
screen->initialize(window, true);
```
These lines give me this error: 
```
"public: __thiscall nanogui::Screen::Screen(void)" (??0Screen@nanogui@@QAE@XZ) referenced in function _main	OpenGLProject
```
This is probably a really simple problem but I don't know much c++ and can't seem to figure this out.
Any help would be appreciated.
Thanks!

**Comment by Petross404** [comment]
Do you mean that you get this error during linking? If yes, you must be missing some linker flags.

**Comment by William-Baker** [comment]
If you're using Visual Studio make sure you've added all the source files to the project directly (so they appear in the solution explorer, use a filter to tidy them up for you). Although I would strongly suggest against using visual studio community & windows. Cross-compiling from Linux, learning CMake and using Ninja will make your life a lot easier when integrating libraries to your project. Apologies if you're using a build system already.

**Comment by nikhilanayak** [comment]
@Petross404 Yeah, during the linking I'm getting these errors. What flags should I be adding.

**Comment by nikhilanayak** [comment]
Actually, I added all the files from nanogui/src. Now, all of the OpenGL functions become undeclared(glCompileShader, glCreateProgram, etc.). Any idea why?
